### PR TITLE
chore: expand scope of adminRole test

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
Improve functional coverage of adminRole e2e test for CDK construct. Now we deploy a lambda whose executionRole is registered as an adminRole, and flex that functionality.

Also added new lambdaInvoke method to e2e tests, and updated `init` command to support flags a bit more consistently.

##### CDK / CloudFormation Parameters Changed
N/A

#### Issue #, if available
N/A

#### Description of how you validated changes
E2E Test

#### Checklist
- [x] PR description included
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by alharris-at** [comment]
E2Es are still running (and a bunch failed for ERR:SocketTimeout), but the admin roles one has passed.

File: packages/amplify-graphql-api-construct-tests/package.json [hunk 1]
```diff
@@ -21,9 +21,11 @@
     "build-tests": "tsc --build tsconfig.tests.json"
   },
   "dependencies": {
+    "@aws-sdk/client-lambda": "^3.438.0",
```
Comment (phani-srikar) [inline packages/amplify-graphql-api-construct-tests/package.json:24]: This is pulling in several additional dependencies with it which could eventually cause the CLI size to increase. I see that we already use `3.338.0` version for the SDK clients including the lambda client. Can we use that version instead?

Comment (alharris-at) [reply packages/amplify-graphql-api-construct-tests/package.json:24]: This shouldn't affect library size, this is is just test package dependencies, I can flip to use 338 though.

**Review by AaronZyLee — COMMENTED** [review]
Looks like the build job fails in CI due to some type issue. Otherwise LGTM

**Review by AaronZyLee — APPROVED** [review]

**Pull request merged**